torchaudio/__init__.py

```diff
@@ -67,36 +91,35 @@ def load(filepath,
         1.
 
     """
-    # stringify if `pathlib.Path` (noop if already `str`)
-    filepath = str(filepath)
-    # check if valid file
-    if not os.path.isfile(filepath):
-        raise OSError("{} not found or is a directory".format(filepath))
-
-    # initialize output tensor
-    if out is not None:
-        check_input(out)
-    else:
-        out = torch.FloatTensor()
-
-    if num_frames < -1:
-        raise ValueError("Expected value for num_samples -1 (entire file) or >=0")
-    if offset < 0:
-        raise ValueError("Expected positive offset value")
 
-    sample_rate = _torch_sox.read_audio_file(filepath,
-                                             out,
-                                             channels_first,
-                                             num_frames,
-                                             offset,
-                                             signalinfo,
-                                             encodinginfo,
-                                             filetype)
+    if get_audio_backend() == "sox":
+        waveform, sample_rate = sox_backend.load(
```

Comment:
A better way to branch would be by doing a conditional import but with the same name
```
if get_audio_backend() == "sox":
  from sox_backend import load
elif get_audio_backend() == "soundfile":
  from _soundfile_backend import load
else:
  raise NotImplementedError

waveform, sample_rate = load(
            filepath,
            out=out,
            normalization=normalization,
            channels_first=channels_first,
            num_frames=num_frames,
            offset=offset,
            filetype=filetype,
        )
```

Reply:
You might be able to save on these if statements if you overwrite the load functions etc. when switching backends, but i'd almost chalk that up under a performance optimization, so it's not necessary yet